src/lsm/groove.zig

```diff
@@ -648,18 +668,18 @@ pub fn GrooveType(
                 };
 
                 if (config.verify) {
-                    // This is checked in prefetch_enqueue()
-                    assert(worker.context.groove.ids.get_cached(id.*) == null);
+                    // This was checked in prefetch_enqueue().
+                    assert(worker.context.groove.ids.lookup_from_memory(worker.context.snapshot, id.*) == null);
                 }
 
                 // If not in the LSM tree's cache, the object must be read from disk and added
                 // to the auxiliary prefetch_objects hash map.
                 // TODO: this LSM tree function needlessly checks the LSM tree's cache a
                 // second time. Adding API to the LSM tree to avoid this may be worthwhile.
-                worker.context.groove.ids.lookup(
+                worker.context.groove.ids.lookup_from_levels(
```

Comment:
I named it `lookup_from_levels()` because it doesn't necessarily hit the disk at all — all of the necessary blocks may already be in the grid's cache. I'd rather find a better name for `lookup_from_memory`, but haven't come up with one yet.

Reply:
Perfect.